### ISSUE TITLE
Make the package installable via pip and conda

### ISSUE THREAD

**Issue by tzdwi** [project-wavelets/JazzHands#20]
We have docs and a useable API. We should try to put together a release so others can install the package!

**Comment by athulvis** [comment]
What is the status of this issue?

**Comment by tzdwi** [comment]
Looks like we're running into some bugs with the version number when deploying to pip. I've added a `__version__` attribute to `jazzhands/__init__.py` but it's still having some problems. @avivajpeyi, have you run into this before?

**Comment by avivajpeyi** [comment]
Hi all -- Sorry! Didnt notice that #22 was already merged!! Will look into this :D

**Comment by avivajpeyi** [comment]
Hi all, the issue with the release is that there is already a pypi project called [`jazzhands`]!! We can either
1. Change the name jazzhands --> something else :( 
2. Try to [request a name transfer](https://www.python.org/dev/peps/pep-0541/#how-to-request-a-name-transfer)

The name transfer process might take some time -- we would need to contact the owner of [`jazzhands`], and then go through `pypi` to get that name. I think it is doable becuase it seems like [`jazzhands`] is not in use!  

The alternative, changing this `pypi` project name may be an easier path... [This website may help pick a name.](https://acronymify.com/search?q=Weighted+Wavelet+Z-transform)

Thoughts? 



[`jazzhands`]: https://pypi.org/project/jazzhands/0/

**Comment by tzdwi** [comment]
I remember some conversation in slack about using a slightly different name in `pypi`; `jazz-hands` instead of `jazzhands`. It means users will have to `pip install jazz-hands` and then `import jazzhands`, similar to [Everest/everest-pipeline](https://pypi.org/project/everest-pipeline/) but it's not the end of the world...

**Comment by avivajpeyi** [comment]
If someone approves the PR above then I can try testing the release again! 

Although I am a bit concerned about the unit tests failing...

**Comment by avivajpeyi** [comment]
I looked into the unittest errors:
## Numpy not found
Initially when I just added `name='jazz-hands'` I got an error with `numpy not found`:
### DOCTEST:
```
024 .. code-block:: python
025 
026     >>> from jazzhands import *
UNEXPECTED EXCEPTION: ModuleNotFoundError("No module named 'numpy'",)
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/doctest.py", line 1330, in __run
    compileflags, 1), test.globs)
  File "<doctest getting_started.rst[0]>", line 1, in <module>
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/jazz_hands-0.1.dev1+geb65da3-py3.6.egg/jazzhands/__init__.py", line 8, in <module>
    from jazzhands.wavelets import *  # noqa
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/jazz_hands-0.1.dev1+geb65da3-py3.6.egg/jazzhands/wavelets.py", line 6, in <module>
    import numpy as np
ModuleNotFoundError: No module named 'numpy'
```

### UNITTEST:
```
Using /opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages
Finished processing dependencies for jazz-hands==0.1.dev1+geb65da3
ImportError while loading conftest '/home/runner/work/JazzHands/JazzHands/jazzhands/conftest.py'.
jazzhands/__init__.py:8: in <module>
    from jazzhands.wavelets import *  # noqa
jazzhands/wavelets.py:6: in <module>
    import numpy as np
E   ModuleNotFoundError: No module named 'numpy'
```

## extension_helpers not found

 I added numpy to the `install-requires` (even though its present in the setup.cfg), and now I'm getting a new error: 

```
Adding jazz-hands 0.1.dev1+g6344467 to easy-install.pth file
Installing astropy-package-template-example script to /opt/hostedtoolcache/Python/3.6.12/x64/bin
..
Processing astropy-4.1rc2.tar.gz
Writing /tmp/easy_install-g_y2_d5n/astropy-4.1rc2/setup.cfg
Running astropy-4.1rc2/setup.py -q bdist_egg --dist-dir /tmp/easy_install-g_y2_d5n/astropy-4.1rc2/egg-dist-tmp-kx9mfkyn
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/setuptools/sandbox.py", line 154, in save_modules
    yield saved
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/setuptools/sandbox.py", line 195, in setup_context
    yield
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/setuptools/sandbox.py", line 250, in run_setup
    _execfile(setup_script, ns)
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/setuptools/sandbox.py", line 45, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-g_y2_d5n/astropy-4.1rc2/setup.py", line 102, in <module>
ModuleNotFoundError: No module named 'extension_helpers'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 85, in <module>
    'write_to_template': VERSION_TEMPLATE},
 ...
  File "/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/site-packages/setuptools/sandbox.py", line 45, in _execfile
    exec(code, globals, locals)
  File "/tmp/easy_install-g_y2_d5n/astropy-4.1rc2/setup.py", line 102, in <module>
ModuleNotFoundError: No module named 'extension_helpers'
Error: Process completed with exit code 1.
```

I suspect that this is something more than just adding `extension_helpers` to the `install_requires`.

**Comment by avivajpeyi** [comment]
Alright, something quite weird is up -- just changing the name has caused some issues with the installation... hmmm

**Comment by theand9** [comment]
I saw #31 and you have added `extension_helpers` to the _install_requires_ section what's the error now?

**Comment by avivajpeyi** [comment]
The main change we want is just `name=jazzhands`  --> `name=jazz-hands`.

So, I dont think we should need to add `extension_helpers` to the install_requires. That package should get automatically installed with `pip install astropy`, as it was when the project's `name=jazzhands` (back then the CI was able to install `jazzhands` fine without adding `extension_helpers` to the requirements).

Hence, I think there is something weirder going on, that might go away when we keep adding stuff to `install_requires`, but might cause some issues down the line. (Eg now `python setup.py install` on the CI isnt picking up `jinja2` which could also be added to `install-requires` even though we didnt need to add it to `install-requires` when `name=jazzhands`)

**Comment by avivajpeyi** [comment]
Also, you can view the current errors by checking the details on the [most recent CI job for the PR](https://github.com/project-wavelets/JazzHands/pull/31)

**Comment by avivajpeyi** [comment]
Hey all, going to try just doing it manually to get a release asap

**Comment by avivajpeyi** [comment]
Manual releasing (and the online releasing) both face the same issues about the name. This is the same issue discussed [here](https://github.com/pypa/setuptools_scm/issues/342).

```
python setup.py sdist
...
jazz-hands-0.1.dev108+g5d1fce9.tar.gz 
...
HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/
'0.1.dev108+g5d1fce9' is an invalid value for Version. Error: Can't use PEP 440 local versions. See https://packaging.python.org/specifications/core-metadata for more information.
```

**Comment by avivajpeyi** [comment]
Closed with #33 #31 #22
